rootex/framework/scene.h

```diff
@@ -80,6 +84,7 @@ class Scene
 	bool removeChild(Scene* toRemove);
 
 	void setName(const String& name);
+	void setm_BaseID(const SceneID& input);
```

Comment:
Rename to `setBaseID` and define it here itself
Do we really need an argument here?
We can directly access BaseID to set m_BaseID

Reply:
m_BaseID is the value stored in json (named so following naming convention of other variables in json) which is updated later after baseid is defined and taken as input.Hence i had to create a function which is called only after baseid gets set, otherwise json stores jargon values